src/relay/transforms/merge_composite.cc

```diff
@@ -66,6 +66,31 @@ class MergeCompositeWrapper : public ExprMutator {
     return root;
   }
 
+  Expr ExtractPattern(const TupleGetItem& pattern, const Expr& root,
+      Map<std::string, Array<Expr>>* var_map, Map<Expr, Expr>* call_map) {
+    if (!root->IsInstance<TupleGetItemNode>()) {
+      return Expr();
+    }
+    auto root_node = Downcast<TupleGetItem>(root);
+    if (pattern->index != root_node->index) {
+      return Expr();
+    }
+    if (pattern->tuple->IsInstance<CallNode>() &&
+        root_node->tuple->IsInstance<CallNode>()) {
+      Expr new_arg;
+      if (call_map->find(pattern->tuple) != call_map->end()) {
+        new_arg = (*call_map)[pattern->tuple];
+      } else {
+        new_arg = ExtractPattern(Downcast<Call>(pattern->tuple),
```

Comment:
I'm not sure I understand why there's a downcast to a Call here, shouldn't it be a Tuple?

Reply:
Thanks for reviewing!
`pattern->tuple` can be a tuple or it can be any node with a TupleType output. In this call it is a Call node (batch_norm) which has multiple outputs so the output is a TupleType.

Reply:
Ah, I get it now, thanks.